GitHub workflow for actions/stale to label and close stale issues

jeffhandley opened the pull request:
This adopts [actions/stale](https://github.com/actions/stale) into a new workflow to label and close stale issues, with a configuration achieving the following behavior:

* Applies only to issues, and not PRs
* Labels issues as https://github.com/dotnet/sdk/labels/stale after `2192` days (6 years), and leaves a comment when doing so
* Removes the label if an issue is updated or commented
    * This is the default behavior for [`remove-stale-when-updated`](https://github.com/actions/stale?tab=readme-ov-file#remove-stale-when-updated)
    * This does not happen immediately, but will be picked up during a subsequent run of the workflow
* Closes stale issues after `30` more days of inactivity, and leaves a comment when doing so
* Runs the workflow every hour, with a limit of `500` operations per run (up from the default of `30` but low enough to avoid hitting the rate limit)
* Provides a `workflow_dispatch` to allow the workflow to be run interactively in the GitHub UI with input
* Only run on the dotnet org (not forks)

For this to successfully run, the following [GitHub Actions settings](https://github.com/dotnet/sdk/settings/actions) must be enabled:

1. Allow enterprise, and select non-enterprise, actions and reusable workflows
2. Allow actions created by GitHub

This replaces [Add repository policy to label and close stale issues (#40226)](https://github.com/dotnet/sdk/pull/40226).

jeffhandley committed:
GitHub workflow for actions/stale to label and close stale issues

jeffhandley committed:
Run the stale action in ascending issue order

jeffhandley committed:
Set an operations-per-run limit of 500 and allow manual override

MiYanni reviewed (approved):
Looks like the action permissions are set properly. So, this looks good to me!

jeffhandley commented:
Thanks for the collaboration on this, @MiYanni, @marcpopMSFT, and @danmoseley. I ran the workflow interactively and it succeeded for the first run and marked 125 issues as stale. It'll continue at :19 past the hour each hour.

```
Processed issues: 296
New stale issues: 125
Added issues labels: 125
Added issues comments: 125
Fetched items: 300
Fetched items events: 125
Fetched items comments: 125
Operations performed: 503
Github API rate used: 503
Github API rate remaining: 14496; reset at: Thu Apr 25 2024 19:45:06 GMT+0000 (Coordinated Universal Time)
state: persisting info about 296 issue(s)
```

marcpopMSFT commented:
Looks good so far. I scanned through them and it appears to be working as expected. One interesting note is that @jcouv added a label to one issue but it didn't remove the stale label. I'm not sure if he was intending for that to change or not but the action only processes based on new comments, correct?
